--- ../repo/sage/src/bin/sage-env	2023-03-30 15:47:17
+++ sage-env	2023-03-31 12:08:57
@@ -110,9 +110,29 @@
 # Make sure that SAGE_ROOT is either an absolute physical directory name
 # or empty.
 if [ -n "$SAGE_ROOT" ]; then
-    export SAGE_ROOT=$(cd "$SAGE_ROOT" 2>/dev/null && pwd -P)
+    NEW_SAGE_ROOT="$SAGE_ROOT"
+elif [ -f sage -a -d build ]; then
+    NEW_SAGE_ROOT="."
+elif [ -f ../../sage -a -d ../../build ]; then
+    NEW_SAGE_ROOT="../.."
 fi
 
+if [ -n "$NEW_SAGE_ROOT" ]; then
+    # Make NEW_SAGE_ROOT absolute
+    NEW_SAGE_ROOT=`cd "$NEW_SAGE_ROOT" && pwd -P`
+
+    # Warn if NEW_SAGE_ROOT does not equal the old SAGE_ROOT
+    if [ "$SAGE_ROOT" != "$NEW_SAGE_ROOT" -a -n "$SAGE_ROOT" ]; then
+        # But don't warn if we are just dereferencing the symlink
+        if ! [[ -L "$SAGE_ROOT" ]]; then
+            echo >&2 "Warning: overwriting SAGE_ROOT environment variable:"
+            echo >&2 "Old SAGE_ROOT=$SAGE_ROOT"
+            echo >&2 "New SAGE_ROOT=$NEW_SAGE_ROOT"
+        fi
+    fi
+
+fi
+
 # Don't execute the commands more than once for the same version of
 # sage-env... for the same combination of SAGE_LOCAL and SAGE_VENV.
 # "6" indicates the version of the format of the value of SAGE_ENV_VERSION.
@@ -123,6 +143,10 @@
 fi
 export SAGE_ENV_SOURCED="$SAGE_ENV_VERSION"
 
+if [ -n "$NEW_SAGE_ROOT" ]; then
+    export SAGE_ROOT="$NEW_SAGE_ROOT"
+fi
+
 # The compilers are set in order of priority by
 # 1) environment variables
 # 2) compiler installed by sage
@@ -591,6 +615,8 @@
     MAKE=`echo "$MAKE" | sed 's/ .*//'`
 fi
 export MAKE
+
+export PIP_FORMAT="columns"
 
 # Set the cysignals crash logs directory
 if [ -z "$CYSIGNALS_CRASH_LOGS" ]; then
