*** OpenBLAS-0.3.29/utest/Makefile	Sun Jan 12 08:02:02 2025
--- OpenBLAS-0.3.29.new/utest/Makefile	Mon May  5 12:18:50 2025
***************
*** 73,78 ****
--- 73,86 ----
  
  $(UTESTEXTBIN): $(OBJS_EXT)
  	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ ../$(LIBNAME) $(EXTRALIB)
+ else ifeq ($(OSNAME), Darwin)
+ GF_PREFIX := $(shell gfortran -v -E 2>&1 | awk '/^Configured/' | cut -f4 -d' ' | cut -f2 -d=)
+ $(UTESTBIN): $(OBJS)
+ 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ ../$(LIBNAME) $(EXTRALIB) $(FEXTRALIB)
+ 	install_name_tool -add_rpath $(GF_PREFIX)/lib $@
+ $(UTESTEXTBIN): $(OBJS_EXT)
+ 	$(CC) $(CFLAGS) $(LDFLAGS) -L$(GF_PREFIX)/lib -o $@ $^ ../$(LIBNAME) $(EXTRALIB)
+ 	install_name_tool -add_rpath $(GF_PREFIX)/lib $@
  else
  $(UTESTBIN): $(OBJS)
  	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ ../$(LIBNAME) $(EXTRALIB) $(FEXTRALIB)
